float64 t_m_fl
float64 t_m_fr
float64 t_m_rl
float64 t_m_rr
float64 del_s
